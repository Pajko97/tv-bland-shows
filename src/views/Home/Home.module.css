.home_wrapper {
    margin:auto 10rem;
    padding-top:0;
    font-family: 'Red Hat Display';
}

.content {
    margin-top:-10rem;
}
.offset {
    background-color:#EDEDED;
    width:100vw;
    margin-left:-10rem;
}

.offset p {
    margin:0;
    margin-left:10rem;
    margin-right: 10rem;
    padding-bottom: 10rem;
    max-width: 30rem;
}
.show_list {
    display: flex;
    flex-wrap: wrap;
    grid-gap: 1.5rem;
    justify-content: flex-start;
    align-items: flex-start;
}


.home_wrapper h1 {
    margin-top: 2rem;
}





@media only screen and (max-width: 800px) {
    .home_wrapper {
        padding:0;
        margin:auto 2rem;
    }

    .show_list {
        flex-direction: row;
        align-items: center;
        justify-content: space-around;
    }

   
    
}