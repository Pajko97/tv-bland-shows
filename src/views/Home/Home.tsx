import React, { useEffect, useState } from "react";
import API from "../../api/shows-api";
import ShowCard from "../../components/Home/ShowCard/ShowCard";
import Schedule from "../../models/Schedule";
import style from "./Home.module.css";
function Home() {
  const [shows, setShows] = useState<Array<Schedule>>([]);

  useEffect(() => {
    let api = new API();
    api.getLatestShows().then((res) => setShows(res.data));
  }, []);

  return (
    <div className={style.home_wrapper}>
      <div className={style.offset}>
        <p>
          TV Show and web series database. Create personalized schedules.
          Episode guide, cast, crew and character information.
        </p>
      </div>
      <div className={style.content}>
        <h1 className={style.headline}>Last added shows</h1>
        <div className={style.show_list}>
          {shows &&
            shows.map((item: Schedule, i) => {
              return <ShowCard key={i} data={item.show} />;
            })}
        </div>
      </div>
    </div>
  );
}

export default Home;
