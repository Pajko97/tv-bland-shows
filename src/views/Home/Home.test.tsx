import React from "react";
import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import Home from "./Home";

it('renders home', async () => {
        let render;
          render = renderer.create(
              <BrowserRouter>
                <Home/>
              </BrowserRouter>
          ).toJSON();
      
      expect(render).toMatchSnapshot();

});
