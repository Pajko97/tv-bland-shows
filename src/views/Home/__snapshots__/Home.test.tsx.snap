// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders home 1`] = `
<div
  className="home_wrapper"
>
  <div
    className="offset"
  >
    <p>
      TV Show and web series database. Create personalized schedules. Episode guide, cast, crew and character information.
    </p>
  </div>
  <div
    className="content"
  >
    <h1
      className="headline"
    >
      Last added shows
    </h1>
    <div
      className="show_list"
    />
  </div>
</div>
`;
